sdk/clientcore/core/src/main/java/io/clientcore/core/serialization/xml/XmlReadValueCallback.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.serialization.xml;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * A callback used when reading an XML value, such as {@link XmlReader#getNullableElement(XmlReadValueCallback)}.
+ *
+ * @param <T> Input type of the callback.
+ * @param <R> Output type of the callback.
+ */
+@FunctionalInterface
+public interface XmlReadValueCallback<T, R> {
```

Comment:
Should we use `Function` here to match the JDK types (similar to `Consumer` above)? 
`XmlReadValueFunction`.

Reply:
Do you mean `implement Function`? If so, that can't be done as this adds a throws for `XMLStreamException` to the functional method, allowing for all the try-catches to runtime exceptions to be removed in lambda code.

Reply:
No, I was just suggesting updating the name to use the suffix `Function` to indicate that there is an input and an output similar to [Function](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/function/Function.html).

Updated name - `XmlReadValueFunction`